akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/EventSourcedBehaviorTestKitImpl.scala

```diff
@@ -150,10 +132,74 @@ import akka.persistence.typed.internal.EventSourcedBehaviorImpl
       replyProbe.stop()
     }
 
+    val newState = getState()
+    val newEvents = getEvents(oldEvents.size)
+
+    postCommandCheck(newEvents, newState, Some(reply))
+
+    CommandResultImpl[Command, Event, State, R](command, newEvents, newState, Some(reply))
+  }
+
+  override def runCommandExpectNoReply[R](creator: ActorRef[R] => Command): CommandResult[Command, Event, State] = {
```

Comment:
I find it strange that a `noReply` is requiring a `ActorRef[R] => Command`. 
The case in which I don't expect a reply is when my command does not have a `replyTo` actor and my production code will be using a `tell`.

If an user is modelling the command to have a `replyTo` actor, but they ignore it, than the contract is being broken. To send such a message, they will need to use an `ask` and the returned `Future` won't complete. 

I think this method should be: 

```scala
override def runCommandExpectNoReply[R](cmd: Command): CommandResult[Command, Event, State]
```

And if I'm really willing to break the protocol contract I have defined myself, I can create my command with an `ignoreRef` instead. 

Note that this is different than an async reply in which from the outside, it behaves exactly the same as a normal command with reply, except that the reply is delayed.

Reply:
Btw, what do you think about `runCommandWithoutReply`?

It's closer to spoken language. 

I see myself saying: _I will run a command without a reply (or that doesn't have a reply)_. 

I will unlikely say: _I will run a command, but I'm not expecting a reply_.

Reply:
giving a second look to it...

Why we need another method? Isn't `def runCommand(command: Command): CommandResult[Command, Event, State]` what we need when using `noReply`? 

So, `runCommand(cmd)` is for `tell` and `runCommand(replyTo => cmd)` is for `ask`. 

That covers all the cases already, no?

Reply:
@renatocaval I think you are right, the `runCommandExpectNoReply` should not be needed.
Either the command has a replyTo and then a reply is expected (async or immediately), or if the command doesn't have a replyTo then the `runCommand(command)` should be used.

@blanchet4forte Can you confirm that `runCommandExpectNoReply` isn't needed.

For some exotic testing of command with replyTo but expecting no reply it would be possible to combine with a TestProbe.
```
val replyTo = createTestProbe[Reply]()
eventSourcedTestKit.runCommand(StrangeCommand(replyTo.ref)
replyTo.expectNoMessage()
```

Reply:
So on my project I didn't need to make use of `runCommandExpectNoReply`. My issue is fixed by the other work that was done in this ticket.

I'm reluctant to say `runCommandExpectNoReply` is not needed however. I agree that not replying when you are using enforcedReplies behavior is breaking the contract.. but the Behavior api lets you do that if you want with an explicit API that is `Effect.noReply`. Granted I think most use cases of `Effect.noReply` will be of the form I'm using it in -- which is there is a reply but its happening asynchronously via something like a `pipeToSelf`. 

All of that said, I think `runCommandExpectNoReply` is potentially useful for testing purposes. I just have not needed it yet myself.

Reply:
Ok, then we will not include it now. Better to add it when there is a concrete need since we don't see a concrete use case for it ourselves yet (and there is also the alternative with the TestProbe that I showed).

Reply:
I think the test probe approach is good enough 👍